base/compiler/abstractinterpretation.jl

```diff
@@ -1057,6 +1058,31 @@ function abstract_call_known(interp::AbstractInterpreter, @nospecialize(f),
     return abstract_call_gf_by_type(interp, f, argtypes, atype, sv, max_methods)
 end
 
+function abstract_call_opaque_closure(interp::AbstractInterpreter, closure::PartialOpaque, argtypes::Vector{Any}, sv::InferenceState)
+    return CallMeta(Any, nothing)
+end
+
+function most_general_argtypes(closure::PartialOpaque)
+    ret = Any[]
+    cc = widenconst(closure)
+    argt = unwrap_unionall(cc).parameters[1]
+    @assert isa(argt, DataType) && argt.name === typename(Tuple)
```

Comment:
Is the idea we'll refuse to form the `PartialOpaque` if this condition doesn't hold?

Reply:
I guess we could just do `[Vararg{Any}]` otherwise. Let me do that.